lib/_stream_readable.js

```diff
@@ -777,8 +777,13 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
     debug('onerror', er);
     unpipe();
     dest.removeListener('error', onerror);
-    if (EE.listenerCount(dest, 'error') === 0)
-      errorOrDestroy(dest, er);
+    if (EE.listenerCount(dest, 'error') === 0) {
+      if (!dest.destroyed) {
+        errorOrDestroy(dest, er);
+      } else {
+        dest.emit('error', er);
```

Comment:
This should likely emit on `process.nextTick()`

Reply:
Why do you think that? The original error was emitted in the same tick.

Reply:
hmm, ok, can revisit this later then